src/server/pegasus_manual_compact_service.h

```diff
@@ -91,7 +91,9 @@ class pegasus_manual_compact_service : public dsn::replication::replica_base
 
     // manual compact state
     std::atomic<bool> _disabled;
+    std::atomic<bool> _time_natural_increase;
     std::atomic<int> _max_concurrent_running_count;
+    std::atomic<uint64_t> _manual_compact_first_day_s;
```

Comment:
Could you please add some comments for the newly added variable?

The logic is too complex if adding these variables, if the current "periodic", "max_concurrent_running" features are not well implemented, I don't mind you to rewrite them or even remove them.